src/uart.rs

```diff
@@ -665,6 +792,217 @@ impl<'d> core::fmt::Write for UartTxDriver<'d> {
     }
 }
 
+fn new_common<UART: Uart>(
+    _uart: impl Peripheral<P = UART>,
+    tx: Option<impl Peripheral<P = impl OutputPin>>,
+    rx: Option<impl Peripheral<P = impl InputPin>>,
+    cts: Option<impl Peripheral<P = impl InputPin>>,
+    rts: Option<impl Peripheral<P = impl OutputPin>>,
+    config: &config::Config,
+) -> Result<(), EspError> {
+    let tx = tx.map(|tx| tx.into_ref());
+    let rx = rx.map(|rx| rx.into_ref());
+    let cts = cts.map(|cts| cts.into_ref());
+    let rts = rts.map(|rts| rts.into_ref());
+
+    #[allow(clippy::needless_update)]
+    let uart_config = uart_config_t {
+        baud_rate: config.baudrate.0 as i32,
+        data_bits: config.data_bits.into(),
+        parity: config.parity.into(),
+        stop_bits: config.stop_bits.into(),
+        flow_ctrl: config.flow_control.into(),
+        rx_flow_ctrl_thresh: config.flow_control_rts_threshold,
+        source_clk: config.source_clock.into(),
+        ..Default::default()
+    };
+
+    esp!(unsafe { uart_param_config(UART::port(), &uart_config) })?;
+
+    esp!(unsafe {
+        uart_set_pin(
+            UART::port(),
+            tx.as_ref().map_or(-1, |p| p.pin()),
+            rx.as_ref().map_or(-1, |p| p.pin()),
+            rts.as_ref().map_or(-1, |p| p.pin()),
+            cts.as_ref().map_or(-1, |p| p.pin()),
+        )
+    })?;
+
+    esp!(unsafe {
+        uart_driver_install(
+            UART::port(),
+            UART_FIFO_SIZE * 2 * (tx.is_some() as i32),
+            UART_FIFO_SIZE * 2 * (rx.is_some() as i32),
+            0,
+            ptr::null_mut(),
+            0,
+        )
+    })?;
+
+    Ok(())
+}
+
+struct Port<'d, P: sealed::AsPort> {
```

Comment:
Another option is instead of creating methods to wrap `Port`, is to make it public (change the name if so), and impl `Deref` and `DerefMut` for `Uart(|Tx|Rx)Driver`.